docs/new_task_guide.md

```diff
@@ -438,6 +438,8 @@ If other tasks on this dataset are already supported:
 
 It is recommended to include a filled-out copy of this checklist in the README.md for the subfolder you are creating, if you have created a new subfolder in `lm_eval/tasks`.
 
+**Finally, please add a short description of your task(s), along with a link to its subfolder in lm_eval/tasks , to [`lm_eval/tasks/README.md`](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/lm_eval/tasks/README.md) so that users can discover your task in the library, and follow the link to your README for more information about the variants supported, their task names, and the original source of the dataset and/or evaluation setup.**
```

Comment:
What actually populates the `--tasks list` result? I'm wondering if this contributing guide aught to reference that as well when adding to the below README, though maybe it happens automatically somehow?

Reply:
The `--tasks list` result is populated with `task_manager.all_tasks` !

Future updates to this should: 
- make it possible to access separately a list of all tasks, and all groups (say via attributes like `task_manager.all_groups` and `task_manager.all_subtasks`, or updating `task_manager.all_tasks` to exclude groupings)
- have the output from `--task list` be rendered as a markdown table, somewhat similar to this lm_eval/tasks/README.md file contents
- probably move the logic that adds newlines to `task_manager.all_tasks` into a `TaskManager.list_all_tasks()` method